### PR TITLE
[[ Bug 22942 ]] Ensure the shellcommand defaults to COMSPEC on Windows

### PR DESCRIPTION
**Pull request by livecodepanos**
Goes together with https://github.com/livecode/livecode-ide/pull/2145

### CODE REVIEW

File: engine/src/dskw32.cpp [hunk 1]
```diff
@@ -1534,8 +1534,8 @@ struct MCWindowsDesktop: public MCSystemInterface, public MCWindowsSystemService
 			}
 		}
 
-		// On NT systems 'cmd.exe' is the command processor
-		MCValueAssign(MCshellcmd, MCSTR("cmd.exe"));
+		// On NT systems use the env var COMSPEC (= C:\Windows\System32\cmd.exe)
```
Comment (runrevmark) [inline engine/src/dskw32.cpp]:
```
/* Default the shellCommand to the value of the COMSPEC environment
 * variable. */
```

File: engine/src/dskw32.cpp [hunk 2]
```diff
@@ -1534,8 +1534,8 @@ struct MCWindowsDesktop: public MCSystemInterface, public MCWindowsSystemService
...
+		MCS_getenv(MCSTR("COMSPEC"), MCshellcmd);
```
Comment (runrevmark) [inline engine/src/dskw32.cpp]: So we should use MCValueAssign here like elsewhere:
```
MCAutoStringRef t_comspec;
if (MCS_getenv(MCSTR("COMSPEC"), &t_comspec))
{
    MCValueAssign(MCshellcmd, *t_comspec);
}
else
{
    MCValueAssign(MCshellcmd, kMCEmptyString);
}

File: docs/dictionary/property/shellCommand.lcdoc [hunk 3]
```diff
@@ -22,8 +22,8 @@ set the shellCommand to "/bin/sh/ksh"
 Value:
 The <shellCommand> is a string.
 By default, the <shellCommand> <property> is set to "/bin/sh" (the
-Bourne shell) on <Unix|Unix systems>, and "command.com" on Windows
-systems. 
+Bourne shell) on <Unix|Unix systems>, and to the environment variable "COMSPEC" on Windows
```
Comment (runrevmark) [inline docs/dictionary/property/shellCommand.lcdoc]:
```
and to the value of the COMSPEC environment variable on Windows systems.
```
(I don't think there's a need to say anymore than that).

Should Windows be referred to like `<Windows|Windows systems>` as it is for Unix?

File: docs/notes/bugfix-22942.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# Ensure the shellcommand defaults to COMSPEC on Windows
```
Comment (runrevmark) [inline docs/notes/bugfix-22942.md]: `shellCommand`

**Review by runrevmark — COMMENTED** [review]
With this patch there is the possibility that `the shellCommand` could be empty. There is no way to correctly guess what to use if COMSPEC is empty so we should make sure that shell() does not work if it is (indeed, this would be sensible on all platforms!)

In `MCFilesEvalShell` after  the secureMode check do:
```
if (MCStringIsEmpty(MCshellcmd))
{
    MCeerror->add(EE_SHELL_BADCOMMAND, 0, 0, "no shell");
    ctxt . Throw();
    return;
}
````

This will cause an error to be thrown if the shellCommand is empty - rather than it trying to run launch a process with a malformed command line (which will probably fail anyway, but this stops the attempt!).

**Review by runrevmark — APPROVED** [review]

**Comment by runrevmark** [comment]
@livecode-vulcan review ok d8cb11f

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok d8cb11f12c2e196d1d81c4036e60b6f5d74af0a4

**Comment by livecode-vulcan** [comment]
😎 test success d8cb11f12c2e196d1d81c4036e60b6f5d74af0a4

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/895)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/304)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/304)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/304)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/304)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2421)
* try-community-universal-ios-iphoneos14.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.0/builds/23)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/74)
* try-community-universal-ios-iphonesimulator14.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.0/builds/24)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/70)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2448)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2437)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2436)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2471)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2411)

**Pull request merged**